tests/test_features.py

```
@@ -83,7 +83,8 @@ def test_build_snapshot_as_of_tracks_latest_clean_indicator_row() -> None:
 
     snapshot = build_snapshot(frame, symbol="CLEAN", interval="1d", lookback="80d")
 
-    assert snapshot.as_of == index[-2].isoformat()
+    expected_as_of = index.to_list()[-2]
+    assert snapshot.as_of == str(getattr(expected_as_of, "isoformat")())
```

Ruff (0.15.12)

[warning] Do not call `getattr` with a constant attribute value. It is not any safer than normal property access.

Replace `getattr` with attribute access

(B009)